### PR TITLE
Add changelog message about destructive actions

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Documentation

### Description

Add missed changelog for user-facing changes to kroxylicious-api

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Review by k-wall — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +26,10 @@ threads, then switch back to an execution context where mutation of Filter membe
 * Record Encryption Filter: Data Encryption Keys will now be refreshed one hour after creation by default.
 This is a bugfix for [#1139](https://github.com/kroxylicious/kroxylicious/issues/1339) to ensure we start
 using new key material after key-encryption-keys are rotated in the KMS within some controlled duration.
+* Moved Deprecated classes out of kroxylicious-api to internal kroxylicious-runtime:
```
Comment (SamBarker) [inline CHANGELOG.md:29]: IIRC in previous releases we've had a deprecated/removed section. I think we should do that here as well to make the removal from the public API really stand out.

**Review by SamBarker — APPROVED** [review]
Happy with the content but I'd add a removed header.

File: CHANGELOG.md (hunk 1 above)
Comment (robobario) [reply CHANGELOG.md:29]: This is under the `### Changes, deprecations and removals` zone, but yes I'll change it to be under a more visible banner like:

`Breaking changes to public kroxylicious-api`

To collect all the changes that are coming in other PRs under bullet points

Comment (SamBarker) [reply CHANGELOG.md]: Perfect. Sorry I'd missread the diff I didn't twig it skipped the header.

**Pull request merged**